flow/model/cdc_stream.go

```diff
@@ -32,11 +34,13 @@ func NewCDCStream[T Items](channelBuffer int) *CDCStream[T] {
 	return &CDCStream[T]{
 		records:            make(chan Record[T], channelBuffer),
 		SchemaDeltas:       make([]*protos.TableSchemaDelta, 0),
-		emptySignal:        make(chan bool, 1),
+		emptySignal:        make(chan struct{}),
```

Comment:
is removing buffered intentional?

Reply:
yes, since with this change nothing is ever pushed into the channel, it's only ever closed